src/components/facets/facet-check-list/check-list-item/facet-check-list-item.component.ts

```diff
@@ -1,14 +1,19 @@
 import { FocusableOption } from '@angular/cdk/a11y';
-import { ChangeDetectionStrategy, Component, ElementRef, EventEmitter, Input, Output, ViewChild } from '@angular/core';
+import { ChangeDetectionStrategy, Component, ElementRef, EventEmitter, HostBinding, Input, Output, ViewChild } from '@angular/core';
 import { Facet } from '../../models/facet';
 
+let uniqueId = 0;
+
 @Component({
     selector: 'ux-facet-check-list-item',
     templateUrl: './facet-check-list-item.component.html',
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class FacetCheckListItemComponent implements FocusableOption {
 
+    /** A unique id for the component. */
+    @Input() @HostBinding() id: string = `ux-facet-check-list-item-${++uniqueId}`;
```

Comment:
This isn't a public facing component so you'll have to pass the id through from the parent component (and index it).